Error in benchmarked code should fail benchmark CI

Example:
https://github.com/denoland/deno/pull/4173

`http_bench.ts` errors out but benchmark CI step does not fail. 

https://github.com/denoland/deno/issues/4149

`deno_tcp.ts` segfaults and yet the benchmark CI step does not fail. 

If that was the case it could help us catch bugs described above faster.

This requires to change `run` function in `tools/http_benchmark.py` such that it check error code of spawned server and even better if it did one more request after benchmark is done to ensure that server is still alive.

It looks like `worker_round_robic_bench.ts` suffers the same problem:
```
Compile file:///home/runner/work/deno/deno/cli/tests/workers_round_robin_bench.ts
error TS2532: Object is possibly 'undefined'.

► file:///home/runner/work/deno/deno/cli/tests/workers_round_robin_bench.ts:34:3

34   promise.resolve(data);
     ~~~~~~~
```
https://github.com/denoland/deno/pull/4180/checks?check_run_id=476742985

--- a/tools/http_benchmark.py
+++ b/tools/http_benchmark.py
@@ -186,6 +186,10 @@ def run(server_cmd, port, merge_env=None, origin_cmd=None):
         print output
         return stats
     finally:
+        server_retcode = server.poll()
+        if server_retcode is not None and server_retcode != 0:
+            print "server ended with error"
+            sys.exit(1)
         server.kill()
         if origin is not None:
             origin.kill()

--- a/tools/http_server.py
+++ b/tools/http_server.py
@@ -352,6 +352,8 @@ def spawn():
         yield servers
     finally:
         for s in servers:
+            #Make sure all servers still running, if not assume there was an error
+            assert s.thread.is_alive()
             s.server.shutdown()
 
 

--- a/tools/http_server.py
+++ b/tools/http_server.py
@@ -352,7 +352,8 @@ def spawn():
         yield servers
     finally:
         for s in servers:
-            #Make sure all servers still running, if not assume there was an error
+            # Make sure all servers still running,
+            # if not assume there was an error
             assert s.thread.is_alive()
             s.server.shutdown()
 

